common/utils.sh

```diff
@@ -93,4 +93,100 @@ function __findClosestVersion() {
     
     echo "$selectedVersion"
     return
+}
+
+function __getTotalRam() {
+    # check for free, which is part of gnu core-utils which is 
+    # present on almost every distro, but not macOS
+    if which free > /dev/null; then
+        free --mebi | awk 'NR==2{print $2}'
+        return
+    fi
+    # here we grab the memsize in bytes then convert to MiB
+    # this works on macOS - Big Sur
+    size=$(sysctl hw.memsize | cut -d ' ' -f 2)
+    echo $((size / 1024 / 1024))
+    return
+}
+
+function __convertToMiB() {
```

Comment:
Thankfully you have all those unit tests now so we don't need to look at this! 👍